context: false in stats throws

Following webpack config throws an error:


```js
var webpack = require('webpack');

/** @type {webpack.Configuration} */
module.exports = {
    entry: {
        main: './index.js'
    },
    output: {
        path: 'dist',
        filename: '[name].js'
    },
    stats: {
        context: false, // Removing this fixes it
        colors: true,
        chunkOrigins: false,
        hash: false,
        version: false,
        timings: false,
        assets: false,
        chunks: false,
        chunkModules: false,
        modules: false,
        children: false,
        cached: false,
        reasons: false,
        source: false,
        errorDetails: false,
    },
}
```

**Error**

```

> test@1.0.0 test /Users/mazimi/Desktop/test
> webpack

/Users/mazimi/Desktop/test/node_modules/webpack/lib/RequestShortener.js:11
		directory = directory.replace(/\\/g, "/");
		                      ^

TypeError: directory.replace is not a function
    at new RequestShortener (/Users/mazimi/Desktop/test/node_modules/webpack/lib/RequestShortener.js:11:25)
    at Stats.toJson (/Users/mazimi/Desktop/test/node_modules/webpack/lib/Stats.js:35:28)
    at Stats.toString (/Users/mazimi/Desktop/test/node_modules/webpack/lib/Stats.js:355:20)
    at compilerCallback (/Users/mazimi/Desktop/test/node_modules/webpack/bin/webpack.js:340:31)
    at /Users/mazimi/Desktop/test/node_modules/webpack/lib/Compiler.js:256:15
    at Compiler.emitRecords (/Users/mazimi/Desktop/test/node_modules/webpack/lib/Compiler.js:351:37)
    at /Users/mazimi/Desktop/test/node_modules/webpack/lib/Compiler.js:249:12
    at /Users/mazimi/Desktop/test/node_modules/webpack/lib/Compiler.js:344:11
    at next (/Users/mazimi/Desktop/test/node_modules/tapable/lib/Tapable.js:154:11)
    at Compiler.compiler.plugin (/Users/mazimi/Desktop/test/node_modules/webpack/lib/performance/SizeLimitsPlugin.js:99:4)
    at Compiler.applyPluginsAsyncSeries1 (/Users/mazimi/Desktop/test/node_modules/tapable/lib/Tapable.js:158:13)
    at Compiler.afterEmit (/Users/mazimi/Desktop/test/node_modules/webpack/lib/Compiler.js:341:8)
    at Compiler.<anonymous> (/Users/mazimi/Desktop/test/node_modules/webpack/lib/Compiler.js:336:14)
    at /Users/mazimi/Desktop/test/node_modules/async/dist/async.js:356:16
    at iteratorCallback (/Users/mazimi/Desktop/test/node_modules/async/dist/async.js:936:13)
    at /Users/mazimi/Desktop/test/node_modules/async/dist/async.js:840:16
    at /Users/mazimi/Desktop/test/node_modules/graceful-fs/graceful-fs.js:43:10
    at FSReqWrap.oncomplete (fs.js:123:15)
npm ERR! Test failed.  See above for more details.
```

--- a/schemas/webpackOptionsSchema.json
+++ b/schemas/webpackOptionsSchema.json
@@ -569,20 +569,32 @@
         },
         "exclude": {
           "allOf": [
-            {"$ref": "#/definitions/ruleSet-condition"},
-            {"absolutePath": true}
+            {
+              "$ref": "#/definitions/ruleSet-condition"
+            },
+            {
+              "absolutePath": true
+            }
           ]
         },
         "include": {
           "allOf": [
-            {"$ref": "#/definitions/ruleSet-condition"},
-            {"absolutePath": true}
+            {
+              "$ref": "#/definitions/ruleSet-condition"
+            },
+            {
+              "absolutePath": true
+            }
           ]
         },
         "issuer": {
           "allOf": [
-            {"$ref": "#/definitions/ruleSet-condition"},
-            {"absolutePath": true}
+            {
+              "$ref": "#/definitions/ruleSet-condition"
+            },
+            {
+              "absolutePath": true
+            }
           ]
         },
         "loader": {
@@ -613,8 +625,12 @@
         },
         "resource": {
           "allOf": [
-            {"$ref": "#/definitions/ruleSet-condition"},
-            {"absolutePath": true}
+            {
+              "$ref": "#/definitions/ruleSet-condition"
+            },
+            {
+              "absolutePath": true
+            }
           ]
         },
         "resourceQuery": {
@@ -625,8 +641,12 @@
         },
         "test": {
           "allOf": [
-            {"$ref": "#/definitions/ruleSet-condition"},
-            {"absolutePath": true}
+            {
+              "$ref": "#/definitions/ruleSet-condition"
+            },
+            {
+              "absolutePath": true
+            }
           ]
         },
         "use": {
@@ -876,72 +896,72 @@
           "type": "object",
           "properties": {
             "context": {
-                "type": "string",
-                "description": "context directory for request shortening"
+              "type": "string",
+              "description": "context directory for request shortening"
             },
             "hash": {
-                "type": "boolean",
-                "description": "add the hash of the compilation"
+              "type": "boolean",
+              "description": "add the hash of the compilation"
             },
             "version": {
-                "type": "boolean",
-                "description": "add webpack version information"
+              "type": "boolean",
+              "description": "add webpack version information"
             },
             "timings": {
-                "type": "boolean",
-                "description": "add timing information"
+              "type": "boolean",
+              "description": "add timing information"
             },
             "assets": {
-                "type": "boolean",
-                "description": "add assets information"
+              "type": "boolean",
+              "description": "add assets information"
             },
             "chunks": {
-                "type": "boolean",
-                "description": "add chunk information"
+              "type": "boolean",
+              "description": "add chunk information"
             },
             "chunkModules": {
-                "type": "boolean",
-                "description": "add built modules information to chunk information"
+              "type": "boolean",
+              "description": "add built modules information to chunk information"
             },
             "modules": {
-                "type": "boolean",
-                "description": "add built modules information"
+              "type": "boolean",
+              "description": "add built modules information"
             },
             "children": {
-                "type": "boolean",
-                "description": "add children information"
+              "type": "boolean",
+              "description": "add children information"
             },
             "cached": {
-                "type": "boolean",
-                "description": "add also information about cached (not built) modules"
+              "type": "boolean",
+              "description": "add also information about cached (not built) modules"
             },
             "reasons": {
-                "type": "boolean",
-                "description": "add information about the reasons why modules are included"
+              "type": "boolean",
+              "description": "add information about the reasons why modules are included"
             },
             "source": {
-                "type": "boolean",
-                "description": "add the source code of modules"
+              "type": "boolean",
+              "description": "add the source code of modules"
             },
             "errorDetails": {
-                "type": "boolean",
-                "description": "add details to errors (like resolving log)"
+              "type": "boolean",
+              "description": "add details to errors (like resolving log)"
             },
             "chunkOrigins": {
-                "type": "boolean",
-                "description": "add the origins of chunks and chunk merging info"
+              "type": "boolean",
+              "description": "add the origins of chunks and chunk merging info"
             },
             "modulesSort": {
-                "type": "string",
-                "description": "sort the modules by that field"
+              "type": "string",
+              "description": "sort the modules by that field"
             },
             "chunksSort": {
-                "type": "string",
-                "description": "sort the chunks by that field"
+              "type": "string",
+              "description": "sort the chunks by that field"
             },
             "assetsSort": {
-                "type": "string",
-                "description": "sort the assets by that field"
+              "type": "string",
+              "description": "sort the assets by that field"
             }
           }
         },

--- a/schemas/webpackOptionsSchema.json
+++ b/schemas/webpackOptionsSchema.json
@@ -897,7 +897,8 @@
           "properties": {
             "context": {
               "type": "string",
-              "description": "context directory for request shortening"
+              "description": "context directory for request shortening",
+              "absolutePath": true
             },
             "hash": {
               "type": "boolean",

--- a/schemas/webpackOptionsSchema.json
+++ b/schemas/webpackOptionsSchema.json
@@ -873,7 +873,77 @@
       "description": "Used by the webpack CLI program to pass stats options.",
       "anyOf": [
         {
-          "type": "object"
+          "type": "object",
+          "properties": {
+            "context": {
+                "type": "string",
+                "description": "context directory for request shortening"
+            },
+            "hash": {
+                "type": "boolean",
+                "description": "add the hash of the compilation"
+            },
+            "version": {
+                "type": "boolean",
+                "description": "add webpack version information"
+            },
+            "timings": {
+                "type": "boolean",
+                "description": "add timing information"
+            },
+            "assets": {
+                "type": "boolean",
+                "description": "add assets information"
+            },
+            "chunks": {
+                "type": "boolean",
+                "description": "add chunk information"
+            },
+            "chunkModules": {
+                "type": "boolean",
+                "description": "add built modules information to chunk information"
+            },
+            "modules": {
+                "type": "boolean",
+                "description": "add built modules information"
+            },
+            "children": {
+                "type": "boolean",
+                "description": "add children information"
+            },
+            "cached": {
+                "type": "boolean",
+                "description": "add also information about cached (not built) modules"
+            },
+            "reasons": {
+                "type": "boolean",
+                "description": "add information about the reasons why modules are included"
+            },
+            "source": {
+                "type": "boolean",
+                "description": "add the source code of modules"
+            },
+            "errorDetails": {
+                "type": "boolean",
+                "description": "add details to errors (like resolving log)"
+            },
+            "chunkOrigins": {
+                "type": "boolean",
+                "description": "add the origins of chunks and chunk merging info"
+            },
+            "modulesSort": {
+                "type": "string",
+                "description": "sort the modules by that field"
+            },
+            "chunksSort": {
+                "type": "string",
+                "description": "sort the chunks by that field"
+            },
+            "assetsSort": {
+                "type": "string",
+                "description": "sort the assets by that field"
+            }
+          }
         },
         {
           "type": "boolean"

